src/plugins/routing.js

```diff
@@ -27,23 +27,42 @@ function localePathFactory (i18nPath, routerPath) {
       route = { name: route }
     }
 
-    // Build localized route options
-    let name = route.name + (STRATEGY === STRATEGIES.NO_PREFIX ? '' : routesNameSeparator + locale)
+    let localizedRoute
```

Comment:
nit: maybe `Object.assign` route right away and then just update needed property in individual branches.